scaleway/resource_iam_group_membership.go

```diff
@@ -0,0 +1,162 @@
+package scaleway
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	iam "github.com/scaleway/scaleway-sdk-go/api/iam/v1alpha1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+func resourceScalewayIamGroupMembership() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceScalewayIamGroupMembershipCreate,
+		ReadContext:   resourceScalewayIamGroupMembershipRead,
+		DeleteContext: resourceScalewayIamGroupMembershipDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"user_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The ID of the user",
+				ExactlyOneOf: []string{"application_id"},
+				ForceNew:     true,
+			},
+			"application_id": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "The ID of the user",
+				ExactlyOneOf: []string{"user_id"},
+				ForceNew:     true,
+			},
+			"group_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The ID of the group to add the user to",
+				ForceNew:    true,
+			},
+		},
+	}
+}
+
+func resourceScalewayIamGroupMembershipCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := iamAPI(meta)
+
+	userID := expandStringPtr(d.Get("user_id"))
+	applicationID := expandStringPtr(d.Get("application_id"))
+
+	group, err := api.AddGroupMember(&iam.AddGroupMemberRequest{
+		GroupID:       d.Get("group_id").(string),
+		UserID:        userID,
+		ApplicationID: applicationID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(groupMembershipID(group.ID, userID, applicationID))
+
+	return resourceScalewayIamGroupMembershipRead(ctx, d, meta)
+}
+
+func resourceScalewayIamGroupMembershipRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := iamAPI(meta)
+	groupID, userID, applicationID, err := expandGroupMembershipID(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	group, err := api.GetGroup(&iam.GetGroupRequest{
+		GroupID: groupID,
+	}, scw.WithContext(ctx))
+	if err != nil {
+		return diag.FromErr(err)
```

Comment:
```suggestion
		if is404Error(err) {
			d.SetId("")
			return nil
		}
		return diag.FromErr(err)
```

If the group doesn't exists, neither do the membership